lib/active_merchant/billing/gateways/beanstream.rb

```diff
@@ -191,19 +289,15 @@ def update(vault_id, payment_method, options = {})
         commit(post, true)
       end
 
-      def supports_scrubbing?
-        true
+      private
+      def include_ipp
+        BeanstreamGateway.include IPPCore
       end
 
-      def scrub(transcript)
-        transcript.
-          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
-          gsub(/(&?password=)[^&\s]*(&?)/, '\1[FILTERED]\2').
-          gsub(/(&?trnCardCvd=)\d*(&?)/, '\1[FILTERED]\2').
-          gsub(/(&?trnCardNumber=)\d*(&?)/, '\1[FILTERED]\2')
+      def include_beanstream
+        BeanstreamGateway.include BeanstreamCore
```

Comment:
I don't think this will work as you expect it to. The module will be included on the class, not on only for a specific instance.
Consider using a factory pattern (https://stackoverflow.com/a/41642955/454375)

Reply:
could look something like
```ruby
class BeanstreamGateway < Gateway
  def self.new(options = {})
    region = get_region_from_options(options)
    (region == 1 ? BeanstreamIppGateway : BeanstreamDefautGateway).new(options)
  end
end

class BeanstreamIppGateway < Gateway
  ...
end

class BeanstreamDefautGateway < Gateway
  include BeanstreamCore
  ...
end
```